lgc/patch/PatchCopyShader.cpp

```diff
@@ -294,39 +302,38 @@ void PatchCopyShader::collectGsGenericOutputInfo(Function *gsEntryPoint) {
         Value *output = callInst->getOperand(callInst->getNumArgOperands() - 1); // Last argument
         auto outputTy = output->getType();
 
-        unsigned value = cast<ConstantInt>(callInst->getOperand(0))->getZExtValue();
-        const unsigned streamId = cast<ConstantInt>(callInst->getOperand(2))->getZExtValue();
+        InOutLocationInfo origLocInfo;
+        origLocInfo.setLocation(cast<ConstantInt>(callInst->getOperand(0))->getZExtValue());
+        origLocInfo.setComponent(cast<ConstantInt>(callInst->getOperand(1))->getZExtValue());
+        origLocInfo.setStreamId(cast<ConstantInt>(callInst->getOperand(2))->getZExtValue());
 
-        InOutLocationInfo outLocInfo;
-        outLocInfo.setLocation(value);
-        outLocInfo.setStreamId(streamId);
-
-        auto locInfoMapIt = resUsage->inOutUsage.outputLocInfoMap.find(outLocInfo);
-        if (locInfoMapIt == resUsage->inOutUsage.outputLocInfoMap.end())
+        const auto locInfoMapIt = outputLocInfoMap.find(origLocInfo);
+        if (locInfoMapIt == outputLocInfoMap.end() || visitedLocInfos.count(origLocInfo) > 0)
           continue;
-
-        unsigned location = locInfoMapIt->second.getLocation();
-        const unsigned compIdx = cast<ConstantInt>(callInst->getOperand(1))->getZExtValue();
-
-        unsigned compCount = 1;
-        auto compTy = outputTy;
-        auto outputVecTy = dyn_cast<FixedVectorType>(outputTy);
-        if (outputVecTy) {
-          compCount = outputVecTy->getNumElements();
-          compTy = outputVecTy->getElementType();
+        visitedLocInfos.insert(origLocInfo);
+
+        // Each output call is scalarized and exports 4 bytes for packing
+        unsigned byteSize = 4;
+        auto &newLocByteSizesMap = m_newLocByteSizesMapArray[origLocInfo.getStreamId()];
+        const unsigned newLoc = locInfoMapIt->second.getLocation();
+        if (isPack) {
+          newLocByteSizesMap[newLoc] += byteSize;
+        } else {
+          unsigned compCount = 1;
+          auto compTy = outputTy;
+          auto outputVecTy = dyn_cast<FixedVectorType>(outputTy);
+          if (outputVecTy) {
+            compCount = outputVecTy->getNumElements();
+            compTy = outputVecTy->getElementType();
+          }
+          unsigned bitWidth = compTy->getScalarSizeInBits();
+          // NOTE: Currently, to simplify the design of load/store data from GS-VS ring, we always extend
+          // byte/word to dword and store dword to GS-VS ring. So for 8-bit/16-bit data type, the actual byte size
+          // is based on number of dwords.
+          bitWidth = bitWidth < 32 ? 32 : bitWidth;
```

Comment:
Replace this with std::max(32, bitWidth)

Reply:
Done.